src/b-splines/cubic.jl

```diff
@@ -0,0 +1,235 @@
+"""
+Assuming uniform knots with spacing 1, the `i`th piece of cubic spline
+implemented here is defined as follows.
+
+    y_i(x) = ϕm p(x-i) + ϕ q(x-i) + ϕp q(1- (x-i)) + ϕpp p(1 - (x-i))
+
+where
+
+    p(☆) = 1/6 *(1-☆)^3
+    q(☆) = 2/3 - ☆^2 + 1/2 ☆^3
+
+and the values `ϕX` for `X ⋹ {m, _, p, pp}` are the pre-filtered coefficients.
+
+For future reference, this expands out to the following polynomial:
+
+    y_i(x) = 1/6 cm (1+i-x)^3 + c (2/3 - (x-i)^2 + 1/2 (x-i)^3) +
+             cp (2/3 - (1+i-x)^2 + 1/2 (1+i-x)^3) + 1/6 cpp (x-i)^3
+
+When we derive boundary conditions we will use derivatives `y_0'(x)` and
+`y_0''(x)`
+"""
+Cubic
+
+"""
+Build `rowspec`, `valspec`, `colspec` such that the product
+
+`out = rowspec * valspec * colspec` will be equivalent to:
+
+```julia
+out = zeros(n, n)
+
+for (i, j, v) in args
+    out[i, j] = v
+end
+```
+
+"""
+function _build_woodbury_specs{T}(::Type{T}, n::Int, args::Tuple{Int, Int, Any}...)
```

Comment:
I do agree that this interface is nicer and makes it easier to see what's going on, but it feels a little out-of-place. Perhaps it would fit better in a PR to [WoodburyMatrices.jl](https://github.com/timholy/WoodburyMatrices.jl)?

In the meantime, moving this function to the bottom of the file gets it out of the way from the interpolation-related code.